tools/tsp-client/test/examples/package-sdk-pinning.json

```
@@ -0,0 +1,83 @@
+{
+  "name": "@azure-tools/typespec-python",
+  "version": "0.37.3",
+  "author": "Microsoft Corporation",
+  "description": "Example package.json for tsp-client",
+  "license": "MIT",
+  "type": "module",
+  "main": "dist/src/index.js",
+  "exports": {
+    ".": "./dist/src/index.js",
+    "./testing": "./dist/src/testing/index.js"
+  },
+  "tspMain": "dist/src/index.js",
+  "engines": {
+    "node": ">=14.0.0"
+  },
+  "scripts": {
+    "clean": "rimraf ./dist ./temp ./venv ./node_modules",
+    "build": "tsc -p .",
+    "watch": "tsc -p . --watch",
+    "install": "tsx ./scripts/run-python3.ts ./scripts/install.py",
+    "prepare": "tsx ./scripts/run-python3.ts ./scripts/prepare.py",
+    "lint": "tsx ./scripts/eng/lint.ts",
+    "lint:fix": "eslint . --fix --ext .ts",
+    "format": "npx prettier **/*.ts --write && tsx ./scripts/eng/format.ts",
+    "regenerate": "tsx ./scripts/eng/regenerate.ts",
+    "test": "tsx ./scripts/eng/run-tests.ts"
+  },
+  "files": [
+    "dist/**",
+    "!dist/test/**",
+    "scripts/**",
+    "generator/**"
+  ],
+  "peerDependencies": {
+    "@typespec/compiler": ">=0.63.0 <1.0.0",
+    "@typespec/http": ">=0.63.0 <1.0.0",
+    "@typespec/rest": ">=0.63.0 <1.0.0",
+    "@typespec/versioning": ">=0.63.0 <1.0.0",
+    "@typespec/openapi": ">=0.63.0 <1.0.0",
+    "@azure-tools/typespec-azure-core": ">=0.49.0 <1.0.0",
+    "@azure-tools/typespec-azure-resource-manager": ">=0.49.0 <1.0.0",
+    "@azure-tools/typespec-autorest": ">=0.49.0 <1.0.0",
+    "@azure-tools/typespec-azure-rulesets": ">=0.49.0 <3.0.0",
+    "@azure-tools/typespec-client-generator-core": ">=0.49.1 <1.0.0"
+  },
+  "dependencies": {
+    "js-yaml": "~4.1.0",
+    "semver": "~7.6.2",
+    "tsx": "~4.19.1",
+    "@typespec/http-client-python": "~0.5.1",
+    "fs-extra": "~11.2.0"
+  },
+  "devDependencies": {
+    "@typespec/compiler": "~0.64.0",
+    "@typespec/http": "~0.63.0",
+    "@typespec/rest": "~0.63.0",
+    "@typespec/versioning": "~0.63.0",
+    "@typespec/openapi": "~0.63.0",
+    "@azure-tools/typespec-azure-resource-manager": "~0.49.0",
+    "@azure-tools/typespec-azure-core": "~0.49.0",
+    "@azure-tools/typespec-azure-rulesets": "~0.49.0",
+    "@azure-tools/typespec-autorest": "~0.49.0",
+    "@azure-tools/typespec-client-generator-core": "~0.49.1",
+    "@azure-tools/azure-http-specs": "0.1.0-alpha.4",
+    "@typespec/http-specs": "0.1.0-alpha.5",
+    "@types/js-yaml": "~4.0.5",
+    "@types/node": "~22.5.4",
+    "@types/yargs": "~17.0.33",
+    "@types/semver": "7.5.8",
+    "c8": "^10.1.2",
+    "vitest": "^2.1.2",
+    "rimraf": "~6.0.1",
+    "typescript": "~5.6.3",
+    "typescript-eslint": "^8.8.1",
+    "yargs": "~17.7.2",
+    "chalk": "5.3.0",
+    "@types/fs-extra": "11.0.4"
+  },
+  "azure-sdk/emitter-package-json-pinning": {
+    "@typespec/compiler": ">=0.63.0 <1.0.0"
+  }
```

pinning here would use an array of package names instead of version specs:

```suggestion
  "azure-sdk/emitter-package-json-pinning": [
    "@typespec/compiler"
  ]
```

Also, package name pinning means we don't need the peerDependency above to allow devDependency to be forwarded:
```json
{
  "name": "@azure-tools/typespec-python",
  "version": "0.37.3",
  "peerDependencies": {
    "@typespec/http": ">=0.63.0 <1.0.0",
    "@typespec/rest": ">=0.63.0 <1.0.0"
  },
  "devDependencies": {
    "@typespec/compiler": "~0.64.0",
    "c8": "^10.1.2",
    "vitest": "^2.1.2",
    "rimraf": "~6.0.1"
  },
  "azure-sdk/emitter-package-json-pinning": [
    "@typespec/compiler"
  ]
}

---

This lets us pin transitive peerDependencies without having to restate them

---

Just pushed the update, I noticed this yesterday but decided to wrap it up today. The fixes are in the branch now.

---

Also, this file is a copy/paste of an actual emitter file with some modifications so I think it's fine to leave the peerDeps as a way of ensuring that we are actually taking the values of the azure-sdk/emitter-json-pinning field